SceneFlags.py

```diff
@@ -0,0 +1,39 @@
+from LocationList import location_table
+from math import ceil
+import logging
+
+# Creates a list containing the highest flag used for each scene. 
+# Returns: List of tuples (scene, max_freestanding_flag, max_drop_flag)
+def get_scene_flag_table(world):
+    scene_flags = []
+    for i in range(0, 101):
+        max_freestanding_flag = 0
+        max_drop_flag = 0
+        for location in world.get_locations():
+            if(location.type == "Collectable"):
+                if(location.scene == i):
+                    if("Freestanding" in location.filter_tags or "ActorOverride" in location.filter_tags):
+                        if location.default > max_freestanding_flag:
```

Comment:
The first 2 and second 2 ifs here can just be combined into one, the level of nesting here is a but unnecessary. Same for the one below, which should probably be a multi-line-single-if.

Reply:
Cleaned this up a bit per your recommendation.